src/app.py

```diff
@@ -191,62 +205,76 @@ def update_graph(reports, classes, view, encounter):
-                hovertemplate = "Damage: %{y}<br>Counts: %{customdata}<extra></extra>",
-                marker = dict(color=[color for color in colors]),
-                error_y = dict(
-                    type = 'data',
-                    array = df._std,
-                    thickness = 1.5,
-                    width = 3,
+        
+        # TODO Inform user that some (or all) logs might be missing in the graph
+        if logs:
+
+            logger.info("Calculating average..")
+            t0 = time.time()
+            df = average_logs(logs)
+            t1 = time.time()
+            logger.info('Done calculating average for logs. Took {} s.'.format(t1 - t0))
+
+            class_index = [
+                True if class_ in classes else False for class_ in df['_class']
+            ] if classes else [True] * len(df)
+
+            df = df[class_index].pipe(remove_irrelevant_roles)
+
+            colors = [class_settings[class_]['color'] for class_ in df['_class']]
+
+            figure = go.Figure()
+            figure.add_trace(
+                go.Bar(
+                    x = df.index,
+                    y = df._avg,
+                    customdata = df._counts,
+                    hovertemplate = "Damage: %{y}<br>Counts: %{customdata}<extra></extra>",
+                    marker = dict(color=[color for color in colors]),
+                    error_y = dict(
+                        type = 'data',
+                        array = df._std,
+                        thickness = 1.5,
+                        width = 3,
+                    )
                 )
             )
-        )
-
-        figure.update_layout(
-            template = 'plotly_dark',
-            paper_bgcolor = 'rgba(0, 0, 0, 0)',
-            plot_bgcolor = 'rgba(0, 0, 0, 0)',
-            margin = {'b': 20},
-            bargap = 0.3,
-            hovermode = 'x',
-            autosize = True,
-            title = {
-                'text': f'Percentage of total {view}',
-                'font': {'color': 'white'},
-                'x': 0.5
-            }
-        )
-
-        logger.info("Graph updated.")
-        return dcc.Graph(id='test', figure=figure)
+
+            figure.update_layout(
+                template = 'plotly_dark',
+                paper_bgcolor = 'rgba(0, 0, 0, 0)',
+                plot_bgcolor = 'rgba(0, 0, 0, 0)',
+                margin = {'b': 20},
+                bargap = 0.3,
+                hovermode = 'x',
+                autosize = True,
+                title = {
+                    'text': f'Percentage of total {view}',
+                    'font': {'color': 'white'},
+                    'x': 0.5
+                }
+            )
+
+            logger.info("Graph updated.")
+            return dcc.Graph(id='test', figure=figure)
     return
 
 
+@set_clear_filters_callback(app)
+def clear_page(n_clicks):
+    trigger = dash.callback_context.triggered[0]['prop_id'].split('.')[0]
```

Comment:
feels like it's time to make this into a function, we use this 3 times now, so the rule of three applies :D